services/frontend-service/src/ui/ActionDetails.tsx

```diff
@@ -0,0 +1,298 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2021 freiheit.com*/
+import {
+    DeleteForeverRounded,
+    DeleteOutlineRounded,
+    Error,
+    LockOpenRounded,
+    LockRounded,
+    MoveToInboxRounded,
+} from '@material-ui/icons';
+import * as React from 'react';
+import { BatchAction, LockBehavior } from '../api/api';
+
+export enum ActionTypes {
+    Deploy,
+    PrepareUndeploy,
+    Undeploy,
+    CreateEnvironmentLock,
+    DeleteEnvironmentLock,
+    CreateApplicationLock,
+    DeleteApplicationLock,
+    UNKNOWN,
+}
+
+export type Deploy = {
+    deploy: {
+        environment: string;
+        application: string;
+        version: number;
+    };
+};
+
+export type PrepareUndeploy = {
+    prepareUndeploy: {
+        application: string;
+    };
+};
+
+export type Undeploy = {
+    undeploy: {
+        application: string;
+    };
+};
+
+export type CreateEnvironmentLock = {
+    createEnvironmentLock: {
+        environment: string;
+    };
+};
+
+export type CreateApplicationLock = {
+    createApplicationLock: {
+        environment: string;
+        application: string;
+    };
+};
+
+export type DeleteEnvironmentLock = {
+    deleteEnvironmentLock: {
+        environment: string;
+        lockId: string;
+    };
+};
+
+export type DeleteApplicationLock = {
+    deleteApplicationLock: {
+        environment: string;
+        application: string;
+        lockId: string;
+    };
+};
+
+export type CartAction =
```

Comment:
`CartAction` is the representation within the frontend, while `BatchAction` is the wire format, right? Please explain that in a comment for `CartAction` and ideally for each of the actions create a link to the wire format it represents.

Reply:
That is correct I'll add a comment